### PR TITLE
feat: single step sign in

### PR DESCRIPTION
**Pull request by benbaessler**
## Description

Closes [#207](https://github.com/lenstube-xyz/lenstube/issues/207)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Minor changes in old functionality

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by sasicodes** [comment]
@benbaessler can you please resolve the conflicts 🙇🏻?

**Comment by sasicodes** [comment]
@benbaessler We also moved to monorepo structure!

**Comment by benbaessler** [comment]
@sasicodes applied the repo updates to the code, everything should be working now. 🙂

**Comment by sasicodes** [comment]
@benbaessler 

After successful login, disconnect the wallet directly in Metamask. You will see a Sign in failed error toast.

**Comment by benbaessler** [comment]
> 

The problem is that the `useEffect` hook is being triggered before the variables update on disconnect.
Not sure if this would be in line with best practices, but this is a potential solution: 
```
useEffect(() => {
  if (
    connector?.id &&
    isConnected &&
    chain?.id === POLYGON_CHAIN_ID &&
    !selectedChannel &&
    !selectedChannelId
  ) {
    handleSign()
  }
  // eslint-disable-next-line react-hooks/exhaustive-deps
}, [isConnected])
```
Should I implement this solution @sasicodes ?

**Comment by sasicodes** [comment]
yes, we can try it.

**Comment by benbaessler** [comment]
done, sorry for the delay. @sasicodes

File: apps/web/src/components/Common/Auth/Login.tsx [hunk 1]
```diff
@@ -102,7 +107,31 @@ const Login = () => {
...
+    setChannels,
+    setSelectedChannel,
+    setSelectedChannelId,
+    setShowCreateChannel,
+    signMessageAsync
+  ])
+
+  useEffect(() => {
+    if (
+      connector?.id &&
```
Comment (sasicodes) [inline apps/web/src/components/Common/Auth/Login.tsx:125]: shall we have this all in a const and use it if condition?

**Review by sasicodes — CHANGES_REQUESTED** [review]

**Review by sasicodes — APPROVED** [review]

**Pull request merged**